compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -122,28 +120,48 @@ object Inliner {
   def dropInlined(inlined: Inlined)(implicit ctx: Context): Tree = {
     if (enclosingInlineds.nonEmpty) inlined // Remove in the outer most inlined call
     else {
-      val inlinedAtPos = inlined.call.sourcePos
+      val inlinedAtSpan = inlined.call.span
       val curSource = ctx.compilationUnit.source
 
+      // Tree copier that changes the source of all trees to `curSource`
+      val cpyWithNewSource = new TypedTreeCopier {
+        override protected def sourceFile(tree: tpd.Tree): SourceFile = curSource
+        override protected val untpdCpy: untpd.UntypedTreeCopier = new untpd.UntypedTreeCopier {
+          override protected def sourceFile(tree: untpd.Tree): SourceFile = curSource
+        }
+      }
+
       /** Removes all Inlined trees, replacing them with blocks.
        *  Repositions all trees directly inside an inlined expansion of a non empty call to the position of the call.
        *  Any tree directly inside an empty call (inlined in the inlined code) retains their position.
        */
-      class Reposition extends TreeMap {
-        override def transform(tree: Tree)(implicit ctx: Context): Tree = tree match {
-          case tree: Inlined => transformInline(tree)
-          case _ =>
-            val transformed = super.transform(tree)
-            enclosingInlineds match {
-              case call :: _ if call.symbol.source != curSource =>
-                // Until we implement JSR-45, we cannot represent in output positions in other source files.
-                // So, reposition inlined code from other files with the call position:
-                transformed.withSpan(inlined.call.span)
-              case _ => transformed
-            }
+      class Reposition extends TreeMap(cpyWithNewSource) {
+        override def transform(tree: Tree)(implicit ctx: Context): Tree = {
+          val transformed = tree match {
+            case tree: Inlined => transformInline(tree)
```

Comment:
Why are these treated separately?

Reply:
I believe it was to simplify the change of context while transforming. I will try to manually inline instead.

Reply:
I removed it